src/nal/sps.rs

```diff
@@ -1035,6 +1172,299 @@ impl SeqParameterSet {
 
         Some((timing_info.time_scale as f64) / (2.0 * (timing_info.num_units_in_tick as f64)))
     }
+
+    pub fn pic_width_in_mbs(&self) -> u32 {
+        self.pic_width_in_mbs_minus1 + 1
+    }
+
+    /// From the spec: `PicHeightInMapUnits = pic_height_in_map_units_minus1 + 1`
+    pub fn pic_height_in_map_units(&self) -> u32 {
+        self.pic_height_in_map_units_minus1 + 1
+    }
+
+    /// From the spec: `PicSizeInMapUnits = PicWidthInMbs * PicHeightInMapUnits`
+    pub fn pic_size_in_map_units(&self) -> u32 {
+        self.pic_width_in_mbs() * self.pic_height_in_map_units()
+    }
+}
+
+struct LevelLimit {
+    max_mbps: u32,
+    max_fs: u32,
+    max_dpb_mbs: u32,
+    max_br: u32,
+    max_cpb: u32,
+    max_vmv_r: u32,
+    min_cr: u8,
+    max_mvs_per2mb: Option<NonZeroU8>,
+}
+
+lazy_static! {
```

Comment:
What are your feelings on using `std::sync::LazyLock` (msrp 1.80) or `std::sync::OnceLock` (msrp 1.70) instead? I don't see any particular need to support old Rust versions and personally would rather not have the extra dep.

Alternatively, could use a static sorted array + bisection I suppose.

Reply:
I'll take a look at one of those alternatives in a separate PR and back this block out along with the the `max_val_for_max_dec_frame_buffering()` check for now.